### ISSUE TITLE
Feature request: multi-CDROM autoloading

### ISSUE THREAD

**Issue by kcgen** [dosbox-staging/dosbox-staging#316]
Multi-CDROM games can be dealt with in a couple ways:
1. (best solution) extract all CDROM content to a single directory, and hope the game is smart enough to find what it needs.
2. (2nd best) mount each ISO to a separate CDROM drive letter, and hope the game  automatically iterates through the drives using MSCDEX calls, so it can always find its desired disk without prompting the user. 
3. (worst solution) mount all the ISOs to a single drive letter, and then press `ctrl + F4` as many times as needed when the game asks for a different disk.

In the 3rd scenario, the game compares the ISO's Volume ID or (disk label) against its desired disk's label.   When you press `ctrl + F4` DOSBox "loads" the next ISO, updates the Volume ID, and indicates that the media has changed.  The game again queries the Volume ID and may or may not find a match - wash rinse repeat.

There's an easy way around the 3rd scenario, provided the CDROMs have uniquely-named data files.  This involves copying all of the CDROMs into a single directory (or an "aggregate" directory), and then having DOSBox auto-rotate the Volume IDs when queried.  It only takes one or two queries before the game is fed its desired volume ID, and it carries on looking for the files it needs (which of course are present, because we've copied them all).

The interface for the would look something like:

`mount d cdroms -labels CD1,CD2,CD3,CD4 -t cdrom`

We really only need to add a new `-labels` argument.  A purely hardcoded proof-of-concept is here: https://github.com/dosbox-staging/dosbox-staging/compare/kc/cdrom-labels-1, which is hardcoded to rotate through the labels for the Windows 3.1 game Steven Spielberg's Director's Chair (1996), which uses three CDROMs.  

Previously, the game would popup a full-screen CDROM request page, asking for a particular CDROM.  With this change, the CDROM request just starts rendering and then instantly disappears as the CDROM is quickly "found".

**Comment by kcgen** [comment]
Thanks @GranMiniGun for the following instructions on how to trigger a CD swap using Might and Magic: World of Xeen (CD version):
1. Visit the Travel Pyramid is just outside of town
2. Ask to be transported to Castleview, which will use the Darkside of Xeen CDROM
3. Transporting back to Vetrigo (the starting city) will use the Clouds of Xeen's CDROM

**Comment by johnnovak** [comment]
This in an interesting feature @kcgen! Is there much outstanding work on it?

**Comment by kcgen** [comment]
I've lost that branch, but it was actually a very small change. Thanks for reminding me about it! I've forgotten about it.. but will pull it back in my queue.

**Comment by johnnovak** [comment]
Yeah it's a cool feature. Even virtual disc swapping can get cumbersome 😏

**Comment by Torinde** [comment]
DOSBox-Pure supports that via [.M3U8 files](https://github.com/schellingb/dosbox-pure#loading-m3u8-files)

**Comment by johnnovak** [comment]
Not too sure how many games would benefit from this. I think many games are hardcoded to display the change CD requester anyway. Needs more experimentation if someone decides to implement this.

**Comment by johnnovak** [comment]
The OSD will make switching CD images straightfoward. Until then, Ctrl+F4 will do.
